debian/usr/sbin/kill-old-login-sessions

```
@@ -3,10 +3,13 @@
 limit=86400 # 1 day
 now=$(date +%s)
 
-sessions=$(loginctl list-sessions --no-legend | sed -e 's/^ *//g' | cut -f 1 -d ' ')
-for session in $sessions
-do
-  timestamp=$(loginctl show-session -p Timestamp --value $session)
+sessions=$(loginctl list-sessions -o json | jq -r ".[].session")
+for session in $sessions; do
+  timestamp=$(loginctl show-session -p Timestamp --value $session || true)
```

The session listed above may disappear everywhere in the loop.
`loginctl show-session` and `loginctl kill-session` could also return errors, so it would be good to ignore errors completely, as you have stated in another place.
An error in this program should not be fatal because this is not a mandatory job.

---

> An error in this program should not be fatal

An error is not fatal but it can cause a failure of `kill-old-login-sessions.service` if happens at the end of the loop, and finally it can cause a misjudgment about the machine's health.
One solution is to add `|| true` after every `loginctl` in the loop.
Another is to configure the `.service` unit to ignore an error, though I don't know how.

---

Thank you.
I fixed to continue the loop with ` || continue` when `loginctl` failed.

If the `date` command fails, this script will fail as follows.

```console
$ ./debian/usr/sbin/kill-old-login-sessions 
date: invalid date ‘Fri 2022-09-16 05:41:36 UTC + 1da’
$ echo $?
1
```

```patch
diff --git a/debian/usr/sbin/kill-old-login-sessions b/debian/usr/sbin/kill-old-login-sessions
index 5cf03238..7d700c47 100755
--- a/debian/usr/sbin/kill-old-login-sessions
+++ b/debian/usr/sbin/kill-old-login-sessions
@@ -1,6 +1,6 @@
 #!/bin/sh -e
 
-limit="1day"
+limit="1da"
 now=$(date +%s)
 
 sessions=$(loginctl list-sessions -o json | jq -r ".[].session")
```